api/courses.go

```diff
@@ -465,37 +463,46 @@ func createLecture(c *gin.Context) {
 		playlist = fmt.Sprintf("https://stream.lrz.de/vod/_definst_/mp4:tum/RBG/%s/playlist.m3u8", strings.ReplaceAll(premiereFileName, "-", "_"))
 	}
 
-	lecture := model.Stream{
-		Name:          req.Title,
-		CourseID:      tumLiveContext.Course.ID,
-		LectureHallID: uint(lectureHallId),
-		Start:         req.Start,
-		End:           req.End,
-		StreamKey:     streamKey,
-		PlaylistUrl:   playlist,
-		LiveNow:       false,
-		Recording:     req.Vodup,
-		Premiere:      req.Premiere,
-	}
-	// add file if premiere
-	if req.Premiere || req.Vodup {
-		lecture.Files = []model.File{{Path: fmt.Sprintf("%s/%s", premiereFolder, premiereFileName)}}
+	// Add start date as first event
+	req.DateSeries = append(req.DateSeries, req.Start)
+
+	for _, date := range req.DateSeries {
+		endTime := date.Add(time.Minute * time.Duration(req.Duration))
+
+		lecture := model.Stream{
```

Comment:
In case there are multiple streams created at once, can you do this:

- add a new field to `model.Stream`, `SeriesIdentifier string`, with a default value of `null`
- generate a random string using `uuid.NewV4().String()`
- set this random string on all lectures in an event series

This way we'll be able to update all events that belong to a series in bulk.

Reply:
Current Approach also adds a identifier to single entries. Would you prefer on single entries null as identifier?

Reply:
Single entries should have a null identifier IMO